### DISCUSSION TITLE
Can XMake be used in ROS?

### DISCUSSION THREAD

**Discussion by BlueeHole** [xmake-io/xmake#3657] [Q&A] [answered]
### 你在什么场景下需要该功能？

ROS is a widely used robot operating system. The default build systems catkin and ament of ROS1 and 2 are implemented based on CMake.
Does anyone has the same idea that wants to 'rewrite' catkin in xmake ? 

### 描述可能的解决方案

replace cmake to xmake in ROS build system.

**Comment by waruqi** [comment]
You can try.

**Comment by qlibp** [comment, marked as answer]
I would like to have a try on it, but currently have no idea on how to start.

Basically, the catkin_make use the ROS-provided cmake helper scripts to generate specific file structure and setup.bash files.

Is there a way to call those cmake scripts within xmake? if so, then we could utilize those ROS cmake-helper-scripts to build ROS specific packages via the API from xmake.
